src/HonzaBotner.Discord/Command/Permissions/Data/PermissionEntry.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace HonzaBotner.Discord.Command.Permissions.Data
+{
+    public sealed class PermissionEntry
+    {
+        public Guid Id { get; set; }
+
+        public PermissionEntryType Type { get; set; }
+
+        public PermissionEntryTarget Target { get; set; }
+
+        public ulong TargetId { get; set; }
+
+        public string Permission { get; set; } = null!;
```

Comment:
Since this is concrete class and we have some expectation from its instances and is not bounded by other technology (EF), why use this hack? Just create constructor with all required properties.